wideq/ac.py

```diff
@@ -196,45 +503,145 @@ def poll(self):
+            support_windmode.append(ACMode(option).name)
+    
+        return support_windmode
+
+    @property
+    def support_fanlist(self):
+        """ find device's support fan speed"""
+
+        dict_support_fanmode = self.ac.model.option_item('SupportWindStrength')
+        support_fanmode = []
+        for option in dict_support_fanmode.values():
+            support_fanmode.append(ACWindstrength(option).name)
+    
+        return support_fanmode
+
+    @property
+    def support_swingmode(self):
+        """ find device's support wind direction"""
+
+        dict_support_swingmode = self.ac.model.option_item('SupportWindDir')
+        support_swingmode = []
+        for option in dict_support_swingmode.values():
+            support_swingmode.append(ACSwingMode(option).name)
+    
+        return support_swingmode
+
+    @property
+    def support_pacmode(self):
+        """ find device's support pac model mode"""
+
+        if self.ac.model.model_type == 'PAC':
+            dict_support_pacmode = self.ac.model.option_item('SupportPACMode')
+            support_pacmode = []
+            for option in dict_support_pacmode.values():
+                support_pacmode.append(ACEXTRAMode(option).name)
+        
+        return support_pacmode
+
+    @property
+    def support_racmode(self):
+        """ find device's support rac model mode"""
+
+        if self.ac.model.model_type == 'RAC':
+            dict_support_racmode = self.ac.model.option_item('SupportRACMode')
+            support_racmode = []
+            for option in dict_support_racmode.values():
+                support_racmode.append(ACEXTRAMode(option).name)
+        
+            return support_racmode
+    
+    @property
+    def support_racsubmode(self):
+        """ find device's support rac model submode"""
+
+        if self.ac.model.model_type == 'RAC' or self.ac.model.model_type == 'SAC_CST':
+            dict_support_racsubmode = self.ac.model.option_item('SupportRACSubMode')
+            support_racsubmode = []
+            for option in dict_support_racsubmode.values():
+                support_racsubmode.append(ACRACSubMode(option).name)
+        
+            return support_racsubmode
+
+    @property
+    def support_reservemode(self):
+        """ find device's support reserve mode"""
+
+        dict_support_reservemode = self.ac.model.option_item('SupportReserve')
+        support_reservemode = []
+        for option in dict_support_reservemode.values():
+            support_reservemode.append(ACReserveMode(option).name)
+    
+        return support_reservemode
+
+    @property
+    def support_airpolution(self):
+        """ find device's support air polution mode"""
+
+        dict_support_airpolution = self.ac.model.option_item('SupportAirPolution')
+        support_airpolution = []
+        for option in dict_support_airpolution.values():
+            support_airpolution.append(ACAirPolution(option).name)
```

Comment:
These "support" functions also share a very common structure. Maybe they can be factored out into a single helper function. Using a list comprehension could also make them more readable (and shorter).